### PR TITLE
Add support for coverage through simplecov

### PR DESCRIPTION
**Pull request by MinasMazar**

### CODE REVIEW

File: spec/support/coverage_setup.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'simplecov'
+
+# Save to CircleCI's artifacts directory if we're on CircleCI
+if ENV['CIRCLE_ARTIFACTS']
+  dir = File.join(ENV['CIRCLE_ARTIFACTS'], "coverage")
+  SimpleCov.coverage_dir(dir)
+end
+
+# Evil options :D
```
Comment (mtylty) [inline spec/support/coverage_setup.rb]: @MinasMazar I think you can uncomment everything down here.

File: spec/support/coverage_setup.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# SimpleCov.minimum_coverage 99
+# SimpleCov.minimum_coverage_by_file 90
+# SimpleCov.refuse_coverage_drop
+
+SimpleCov.start
```
Comment (mtylty) [inline spec/support/coverage_setup.rb]: Have you tried running only feature specs to see if SimpleCov works for feature specs? It should be started on all processes running the application (i.e. the commands we launch when running feature specs).

File: pulsar.gemspec [hunk 3]
```diff
@@ -32,4 +31,5 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency 'codeclimate-test-reporter', '~> 0.4'
   gem.add_development_dependency 'rubocop', '~> 0.47'
   gem.add_development_dependency 'timecop', '~> 0.8'
+  gem.add_development_dependency 'simplecov', '~> 0.12.0'
```
Comment (mtylty) [inline pulsar.gemspec]: This is an outdated version of simplecov. Can we move to the latest?

File: spec/support/coverage_setup.rb (hunk 2 above)
Comment (MinasMazar) [inline spec/support/coverage_setup.rb]: Yes, it works also for feature specs.

**Pull request merged**